client/rest-high-level/src/main/java/org/elasticsearch/client/AsyncSearchRequestConverters.java

```diff
@@ -56,4 +55,21 @@ static Request submitAsyncSearch(SubmitAsyncSearchRequest asyncSearchRequest) th
         request.addParameters(params.asMap());
         return request;
     }
+
+    static void addSearchRequestParams(Params params, SubmitAsyncSearchRequest request) {
+        params.putParam(RestSearchAction.TYPED_KEYS_PARAM, "true");
+        params.withRouting(request.getRouting());
+        params.withPreference(request.getPreference());
+        params.withIndicesOptions(request.getIndicesOptions());
+        params.putParam("search_type", request.getSearchType().name().toLowerCase(Locale.ROOT));
+        if (request.getRequestCache() != null) {
+            params.putParam("request_cache", Boolean.toString(request.getRequestCache()));
+        }
+        if (request.getAllowPartialSearchResults() != null) {
+            params.putParam("allow_partial_search_results", Boolean.toString(request.getAllowPartialSearchResults()));
+        }
+        params.putParam("batched_reduce_size", Integer.toString(request.getBatchedReduceSize()));
+        params.putParam("pre_filter_shard_size", Integer.toString(SubmitAsyncSearchRequest.DEFAULT_PRE_FILTER_SHARD_SIZE));
```

Comment:
I think we can remove it. it is not in the spec and we don't allow to set it directly